### PR TITLE
Implemented new recursive functionality.

### PR DESCRIPTION
**Pull request by NissimHadar**
Creates a script in the following format:
```
// This is an automatically generated file, created by auto-tester

var autoTester = Script.require("https://github.com/highfidelity/hifi_tests/blob/master/tests/utils/autoTester.js");
autoTester.enableRecursive();

Script.include("file:///D:/GitHub/hifi-tests-forPR/tests/content/entity/zone/ambientLightInheritance/test.js");
Script.include("file:///D:/GitHub/hifi-tests-forPR/tests/content/entity/zone/create/test.js");

autoTester.runRecursive();
```

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6560/app_12244_PR.apk

File: tools/auto-tester/src/Test.cpp [hunk 1]
```diff
@@ -330,21 +330,22 @@ void Test::createRecursiveScript() {
     }
 
     QTextStream textStream(&allTestsFilename);
-    textStream << "// This is an automatically generated file, created by auto-tester" << endl;
+    textStream << "// This is an automatically generated file, created by auto-tester" << endl << endl;
+
+    textStream << "var autoTester = Script.require(\"https://github.com/highfidelity/hifi_tests/blob/master/tests/utils/autoTester.js\");" << endl;
```
Comment (samcake) [inline tools/auto-tester/src/Test.cpp]: you need to add "?raw=true" at the end of each url to make sure that if it is served from github you get the raw file and not there page created around the file

File: tools/auto-tester/src/Test.cpp [hunk 2]
```diff
@@ -306,8 +306,8 @@ void Test::evaluateTestsRecursively(bool interactiveMode, QProgressBar* progress
     zipAndDeleteTestResultsFolder();
 }
 
-void Test::importTest(QTextStream& textStream, const QString& testPathname, int testNumber) {
-    textStream << "var test" << testNumber << " = Script.require(\"" << "file:///" << testPathname + "\");" << endl;
+void Test::importTest(QTextStream& textStream, const QString& testPathname) {
+    textStream << "Script.include(\"" << "file:///" << testPathname + "\");" << endl;
```
Comment (samcake) [inline tools/auto-tester/src/Test.cpp]: you need to add "?raw=true" at the end of each url to make sure that if it is served from github you get the raw file and not there page created around the file

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6560/HighFidelity-Beta-PR12244-6bb325f6a2d82006b3d82a4c22fdac16309f337c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6560/HighFidelity-Beta-PR12244-6bb325f6a2d82006b3d82a4c22fdac16309f337c.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6561/app_12244_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6561/HighFidelity-Beta-PR12244-f9e0c724ce9962b74d78046747596300482ae206.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6561/HighFidelity-Beta-PR12244-f9e0c724ce9962b74d78046747596300482ae206.exe

**Comment by samcake** [comment]
This work is solely affecting autoTester, all good

**Pull request merged**